src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java

```diff
@@ -21,24 +28,63 @@
  * <http://www.gnu.org/licenses/gpl-3.0.html>.
  * #L%
  */
-
 import org.itsallcode.openfasttrace.importer.*;
 import org.itsallcode.openfasttrace.importer.input.InputFile;
+import org.itsallcode.openfasttrace.importer.tag.config.PathConfig;
+import org.itsallcode.openfasttrace.importer.tag.config.TagImporterConfig;
 
 /**
  * {@link ImporterFactory} for tags in source code files.
  */
 // [impl->dsn~import.full-coverage-tag~1]
-public class TagImporterFactory extends RegexMatchingImporterFactory
+public class TagImporterFactory extends ImporterFactory
 {
-    public TagImporterFactory()
+    private static final Logger LOG = Logger.getLogger(TagImporterFactory.class.getName());
+
+    private static final String DEFAULT_FILE_REGEX = "(?i).*\\.java";
+    private static final Pattern DEFAULT_FILE_PATTERN = Pattern.compile(DEFAULT_FILE_REGEX);
+
+    @Override
+    public boolean supportsFile(final InputFile path)
+    {
+        return supportsDefaultFile(path) || supportsConfiguredFile(path);
+    }
+
+    private boolean supportsConfiguredFile(final InputFile path)
     {
-        super("(?i).*\\.java");
+        return findConfig(path).isPresent();
+    }
+
+    public boolean supportsDefaultFile(final InputFile file)
+    {
+        return DEFAULT_FILE_PATTERN.matcher(file.getPath()).matches();
+    }
+
+    private Optional<PathConfig> findConfig(final InputFile file)
+    {
+        return getPathConfigs()//
+                .peek(c -> LOG.finest(() -> "Checking config " + c + " with file " + file))
+                .filter(config -> config.matches(file)) //
+                .peek(c -> LOG.finest(() -> "Config " + c + " matches file " + file)) //
+                .findFirst();
     }
 
     @Override
-    public Importer createImporter(final InputFile file, final ImportEventListener listener)
+    public Importer createImporter(final InputFile path, final ImportEventListener listener)
+    {
+        if (!supportsFile(path))
+        {
+            throw new ImporterException(
+                    "File '" + path + "' not supported for import. Supported file name patterns: "
+                            + DEFAULT_FILE_REGEX + " and " + getPathConfigs().collect(toList()));
```

Comment:
Actually you need a union here since there might be overlap. Question is if the user-defined config should not completely override the default (i.e. no merging).

Reply:
This exception will never occur as no importer will be created if the file is not supported. This is just an additional assertion and the error message is not important.